public/app/plugins/datasource/prometheus/querybuilder/components/metric-encyclopedia/state/state.ts

```diff
@@ -0,0 +1,245 @@
+import { SelectableValue } from '@grafana/data';
+
+import { HaystackDictionary, MetricsData } from '../types';
+
+export const DEFAULT_RESULTS_PER_PAGE = 100;
+export const MAXIMUM_RESULTS_PER_PAGE = 1000;
+
+/**
+ * The reducer function that uses a switch statement to handle the Metric Encyclopedia actions
+ * @param state
+ * @param action
+ * @returns
+ */
+export function MetricEncyclopediaReducer(state: MetricEncyclopediaState, action: Action): MetricEncyclopediaState {
```

Comment:
if you create your reducer using createSlice from reduxToolkit it will automatically deduce reducer actions from that statement. You can use redux toolkit with useReducer 

s/querybuilder/components/PromQueryBuilderContainer.tsx#L66

Reply:
proper link https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilderContainer.tsx#L66

Reply:
Updated to using redux createSlice.